### PR TITLE
Track balance of due payments to DDC (total for cluster)

### PR DESCRIPTION
**Pull request by evgeny-s**
[Ticket](https://cerenetwork.atlassian.net/browse/CBI-906)
[Specification](https://docs.google.com/document/d/10pkl5HEq9_xMvV7UKrHAvWCwP4BC4-h9O1uioO6S9yA/edit#heading=h.xb6ud5gqvtnd)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -432,30 +435,80 @@ mod ddc {
...
+        pub fn get_total_ddc_balance(&self) -> Balance {
+            self.total_ddc_balance
+        }
+
+        fn set_tier(&mut self, subscription: &mut AppSubscription, new_tier_id: u64) -> Result<()> {
+            let subscription_tier = match self.service_tiers.get(&subscription.tier_id) {
+                None => return Err(Error::TidOutOfBound),
+                Some(v) => v,
+            };
+            Self::actualize_subscription(subscription, subscription_tier);
```
Comment (naure) [inline src/lib.rs:507]: Incorrect, this loses track of `consumed` amount.

File: src/lib.rs [hunk 2]
```diff
@@ -562,23 +615,23 @@ mod ddc {
...
+                None => return Err(Error::NoSubscription),
+                Some(v) => v,
+            };
 
-            self.actualize_subscription(&mut subscription);
+            let subscription_tier = match self.service_tiers.get(&subscription.tier_id) {
+                None => return Err(Error::TidOutOfBound),
+                Some(v) => v,
+            };
+            Self::actualize_subscription(subscription, subscription_tier);
```
Comment (naure) [inline src/lib.rs:627]: Same.

File: src/lib.rs [hunk 3]
```diff
@@ -432,30 +435,80 @@ mod ddc {
...
+                actually_consumed = consumed;
             }
             subscription.last_update_ms = now_ms;
+
+            actually_consumed
         }
 
-        fn set_tier(&mut self, subscription: &mut AppSubscription, new_tier_id: u64) {
-            self.actualize_subscription(subscription);
+        fn actualize_subscription(
```
Comment (naure) [inline src/lib.rs:469]: It’s very important to not forget to track `consumed`. You can add `#[must_use]` to the functions to generate a warning.

https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-must_use-attribute

**Review by naure — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 3 above)
Comment (evgeny-s) [reply src/lib.rs:470]: Added attribute, also fixed all warnings.
Thanks!

File: src/lib.rs (hunk 2 above)
Comment (evgeny-s) [reply src/lib.rs]: Fixed

File: src/lib.rs (hunk 1 above)
Comment (evgeny-s) [reply src/lib.rs]: Fixed

**Comment by evgeny-s** [comment]
@naure , 
Since now we can move `timestamp`, I'm going to add more tests to cover this without isolated timestamp.

**Review by naure — APPROVED** [review]

**Pull request merged**